contracts/application/versioning/FreezableImplementationDirectory.sol

```diff
@@ -4,32 +4,33 @@ import "./ImplementationDirectory.sol";
 
 /**
  * @title FreezableImplementationDirectory
- * @dev Implementation directory which can be made immutable by the owner 
+ * @dev Implementation directory which can be made irreversibly immutable by the owner.
  */
  contract FreezableImplementationDirectory is ImplementationDirectory {
-  // Stores the mutability state of the directory
+  /// @dev Mutability state of the directory.
   bool public frozen;
 
   /**
-   * @dev Modifier that allows functions to be called only when the contract is not frozen
+   * @dev Modifier that allows functions to be called only when the contract is not frozen.
```

Comment:
(Soft) suggestion:
"[...] only before the contract is frozen."?

Reply:
I have (soft) reservations here, but let's go with yours.